tests/integration/eZ/API/RichTextFieldTypeIntegrationTest.php

```diff
@@ -624,6 +626,97 @@ public function testConvertRemoteObjectIdToObjectId($test, $expected)
+     * @throws \ErrorException
+     * @throws \eZ\Publish\API\Repository\Exceptions\BadStateException
+     * @throws \eZ\Publish\API\Repository\Exceptions\ContentTypeFieldDefinitionValidationException
+     * @throws \eZ\Publish\API\Repository\Exceptions\ContentValidationException
+     * @throws \eZ\Publish\API\Repository\Exceptions\InvalidArgumentException
+     * @throws \eZ\Publish\API\Repository\Exceptions\NotFoundException
+     * @throws \eZ\Publish\API\Repository\Exceptions\UnauthorizedException
+     */
+    public function testExternalLinkStoringAfterUpdate(): void
+    {
+        $xmlDocument = $this->createXmlDocumentWithExternalLink();
+        $repository = $this->getRepository();
+        $contentService = $repository->getContentService();
+        $contentTypeService = $repository->getContentTypeService();
+        $locationService = $repository->getLocationService();
+
+        $contentTypeCreateStruct = $contentTypeService->newContentTypeCreateStruct('test-RichText');
+        $contentTypeCreateStruct->mainLanguageCode = 'eng-GB';
+        $contentTypeCreateStruct->names = ['eng-GB' => 'Test'];
+        $contentTypeCreateStruct->creatorId = $repository->getPermissionResolver()->getCurrentUserReference()->getUserId();
+        $contentTypeCreateStruct->creationDate = $this->createDateTime();
+
+        $fieldDefinitionCreateStruct = $contentTypeService->newFieldDefinitionCreateStruct('description', 'ezrichtext');
+        $fieldDefinitionCreateStruct->names = ['eng-GB' => 'Title'];
+        $fieldDefinitionCreateStruct->fieldGroup = 'main';
+        $fieldDefinitionCreateStruct->position = 1;
+        $fieldDefinitionCreateStruct->isTranslatable = true;
+        $contentTypeCreateStruct->addFieldDefinition($fieldDefinitionCreateStruct);
+
+        $contentGroup = $contentTypeService->loadContentTypeGroupByIdentifier('Content');
+        $contentTypeDraft = $contentTypeService->createContentType($contentTypeCreateStruct, [$contentGroup]);
+
+        $contentTypeService->publishContentTypeDraft($contentTypeDraft);
+        $testContentType = $contentTypeService->loadContentType($contentTypeDraft->id);
+
+        $testContentCreateStruct = $contentService->newContentCreateStruct($testContentType, 'eng-GB');
+        $testContentCreateStruct->setField('description', $xmlDocument, 'eng-GB');
+        $content = $contentService->createContent(
+            $testContentCreateStruct,
+            [$locationService->newLocationCreateStruct(2)]
+        );
+        $content = $contentService->publishVersion(
+            $content->versionInfo
+        );
+        $urlIds = $this->getUrlIdsForContentObjectAttributeIdAndVersionNo($content->getField('description')->id, $content->contentInfo->currentVersionNo);
+
+        $contentUpdateStruct = $contentService->newContentUpdateStruct();
+        $contentUpdateStruct->setField('description', $xmlDocument, 'eng-GB');
```

Comment:
> Since the task is about old links, wouldn't it make sense to set XML with a different link and check if old link is no longer present?

Seems this line still causes test not to cover old URLs case.